EnvelopeItem/TransportConfiguration.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Enqueue\MessengerAdapter\EnvelopeItem;
+
+use Symfony\Component\Messenger\EnvelopeItemInterface;
+
+/**
+ * Message envelope item allowing to specify some transport configuration.
+ *
+ * @author Thomas Prelot <tprelot@gmail.com>
+ *
+ * @experimental in 4.1
+ */
+final class TransportConfiguration implements EnvelopeItemInterface
+{
+    /**
+     * @param string $topic
+     */
+    public function __construct(array $configuration)
```

Comment:
Wouldn't providing directly the `?string $topic = null` be better instead of an arbitrary array?

Reply:
Did it for a better time scalability.
An option resolver could also be a possibility.

Reply:
Yup, I believe that it's better for later: we won't have to break the constructor when adding multiple type of parameters.

Reply:
Adding arguments in a constructor is never an issue BC wise (removing/altering is), but fine to me :)

Reply:
Maybe we (at least I) were looking far too much in advance: I saw many additions then too many arguments so the need to refactor it into an array 😄

Reply:
I thought the same :)